app/models/message_draft.rb

```diff
@@ -1,5 +1,23 @@
 # frozen_string_literal: true
 
-# Message drafts are essentially messages with an implied folder (drafts)
+# Message drafts are essentially messages with an implied folder (drafts), and if a reply an implied message
 class MessageDraft < Message
+  validate :check_as_replydraft, if: proc { reply? }
+  validate :check_as_draft, unless: proc { reply? }
```

Comment:
I think we need to pull history and check these validations regardless of the conditions you have provided. We can't count on data coming in to necessarily have gone to the correct endpoint to have these attributes set. We're setting the value of reply? in the controller ourselves based on the endpoint that came in. Instead we should be fetching history and making sure.